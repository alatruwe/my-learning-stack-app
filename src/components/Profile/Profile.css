.profile-form {
  display: flex;
  flex-direction: column;
  margin-top: 20%;
}

/* Desktop view */
@media screen and (min-width: 1024px) {
  .profile-form {
    margin-top: 2%;
  }
}
