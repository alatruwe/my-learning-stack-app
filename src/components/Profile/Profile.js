import React, { Component } from "react";
import { withRouter } from "react-router-dom";
import ValidationError from "../ValidationError/ValidationError.js";

class Profile extends Component {
  constructor(props) {
    super(props);
    this.state = {
      tech1: "",
      tech2: "",
      tech3: "",
    };
  }
  static defaultProps = {
    history: {
      push: () => {},
    },
  };

  validateTech = () => {
    if (
      this.state.tech1 === "" &&
      this.state.tech2 === "" &&
      this.state.tech3 === ""
    ) {
      return "Please chose at least 1 tech you work with";
    }
  };

  updateTech1 = (tech) => {
    this.setState({ tech1: tech });
  };
  updateTech2 = (tech) => {
    this.setState({ tech2: tech });
  };
  updateTech3 = (tech) => {
    this.setState({ tech3: tech });
  };

  handleSubmit = (ev) => {
    ev.preventDefault();
    this.props.history.push(`/dashboard`);
  };

  render() {
    const techError = this.validateTech();
    return (
      <section className="wrapper">
        <h1 className="form-title">
          <span>My Profile</span>
        </h1>

        <div className="form">
          <p>I work with:</p>
          <form className="form-details" onSubmit={this.handleSubmit}>
            <input
              type="text"
              id="tech1"
              name="tech1"
              placeholder="React"
              className="rounded-input"
              value={this.state.tech1}
              onChange={(e) => this.updateTech1(e.target.value)}
            />
            <input
              type="text"
              id="tech2"
              name="tech2"
              placeholder="Javascript"
              className="rounded-input"
              value={this.state.tech2}
              onChange={(e) => this.updateTech2(e.target.value)}
            />
            <input
              type="text"
              id="tech3"
              name="tech3"
              placeholder="Node.js"
              className="rounded-input"
              value={this.state.tech3}
              onChange={(e) => this.updateTech3(e.target.value)}
            />
            {this.validateTech && <ValidationError message={techError} />}
            <div className="form-btn">
              <button
                type="submit"
                className="btn"
                disabled={this.validateTech()}
              >
                Submit
              </button>
            </div>
          </form>
        </div>
      </section>
    );
  }
}

export default withRouter(Profile);
