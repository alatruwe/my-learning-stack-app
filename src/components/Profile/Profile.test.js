import React from "react";
import Profile from "./Profile";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe(`Profile component`, () => {
  it("renders the component by default", () => {
    const wrapper = shallow(<Profile />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
