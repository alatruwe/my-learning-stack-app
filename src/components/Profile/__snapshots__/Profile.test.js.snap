// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile component renders the component by default 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
